Mathlib/Topology/Algebra/Monoid.lean

```diff
@@ -0,0 +1,868 @@
+      _ = map (fun p : M × M => x₀ * p.1 * (p.2 * y₀)) (𝓝 1 ×ᶠ 𝓝 1) :=
+        by simp_rw [uncurry, hleft x₀, hright y₀, prod_map_map_eq, Filter.map_map, Function.comp]
+        -- Porting note: `rw` was able to prove this
+      _ = map ((fun x => x₀ * x) ∘ fun x => x * y₀) (map (uncurry (· * ·)) (𝓝 1 ×ᶠ 𝓝 1)) :=
+        by rw [key, ← Filter.map_map]
+      _ ≤ map ((fun x : M => x₀ * x) ∘ fun x => x * y₀) (𝓝 1) := map_mono hmul
+      _ = 𝓝 (x₀ * y₀) := by rw [← Filter.map_map, ← hright, hleft y₀, Filter.map_map, key₂, ← hleft]
+      ⟩
+#align has_continuous_mul.of_nhds_one ContinuousMul.of_nhds_one
+#align has_continuous_add.of_nhds_zero ContinuousAdd.of_nhds_zero
+
+@[to_additive]
+theorem continuousMul_of_comm_of_nhds_one (M : Type u) [CommMonoid M] [TopologicalSpace M]
+    (hmul : Tendsto (uncurry ((· * ·) : M → M → M)) (𝓝 1 ×ᶠ 𝓝 1) (𝓝 1))
+    (hleft : ∀ x₀ : M, 𝓝 x₀ = map (fun x => x₀ * x) (𝓝 1)) : ContinuousMul M := by
+  apply ContinuousMul.of_nhds_one hmul hleft
+  intro x₀
+  simp_rw [mul_comm, hleft x₀]
+#align has_continuous_mul_of_comm_of_nhds_one continuousMul_of_comm_of_nhds_one
+#align has_continuous_add_of_comm_of_nhds_zero continuousAdd_of_comm_of_nhds_zero
+
+end ContinuousMul
+
+section PointwiseLimits
+
+variable (M₁ M₂ : Type _) [TopologicalSpace M₂] [T2Space M₂]
+
+@[to_additive]
+theorem isClosed_setOf_map_one [One M₁] [One M₂] : IsClosed { f : M₁ → M₂ | f 1 = 1 } :=
+  isClosed_eq (continuous_apply 1) continuous_const
+#align is_closed_set_of_map_one isClosed_setOf_map_one
+#align is_closed_set_of_map_zero isClosed_setOf_map_zero
+
+@[to_additive]
+theorem isClosed_setOf_map_mul [Mul M₁] [Mul M₂] [ContinuousMul M₂] :
+    IsClosed { f : M₁ → M₂ | ∀ x y, f (x * y) = f x * f y } := by
+  simp only [setOf_forall]
+  exact
+    isClosed_interᵢ fun x =>
+      isClosed_interᵢ fun y =>
+        isClosed_eq (continuous_apply _)
+          -- Porting note: proof was:
+          -- `((continuous_apply _).mul (continuous_apply _))`
+          (by continuity)
+#align is_closed_set_of_map_mul isClosed_setOf_map_mul
+#align is_closed_set_of_map_add isClosed_setOf_map_add
+
+-- porting note: split variables command over two lines, can't change explicitness at the same time
+-- as declaring new variables.
+variable {M₁ M₂}
+variable [MulOneClass M₁] [MulOneClass M₂] [ContinuousMul M₂]
+  {F : Type _} [MonoidHomClass F M₁ M₂] {l : Filter α}
+
+/-- Construct a bundled monoid homomorphism `M₁ →* M₂` from a function `f` and a proof that it
+belongs to the closure of the range of the coercion from `M₁ →* M₂` (or another type of bundled
+homomorphisms that has a `MonoidHomClass` instance) to `M₁ → M₂`. -/
+@[to_additive "Construct a bundled additive monoid homomorphism `M₁ →+ M₂` from a function `f`
+and a proof that it belongs to the closure of the range of the coercion from `M₁ →+ M₂` (or another
+type of bundled homomorphisms that has a `add_monoid_hom_class` instance) to `M₁ → M₂`.",
+  simps (config := { fullyApplied := false })]
```

Comment:
I guess this and the next one should also be converted to `(attr := foo)`, but I'm not sure of the exact syntax.

Reply:
`to_additive (attr := simps (config := { fullyApplied := false })) "Construct..."`